backend/services/implementations/userService.ts

```diff
@@ -16,6 +17,50 @@ const getMongoUserByAuthId = async (authId: string): Promise<User> => {
   return user;
 };
 
+/**
+ * Utility method for converting yyyy/mm/dd date
+ *  string to 24-char hex timestamp string
+ *
+ * ObjectID docs:
+ *  https://mongodb.github.io/node-mongodb-native/api-bson-generated/objectid.html
+ * Reference:
+ *  https://kchodorow.com/2011/12/20/querying-for-timestamps-using-objectids/
+ */
+const getDateTime = (date: string): string =>
+  `${Math.floor(new Date(date).getTime() / 1000).toString(16)}0000000000000000`;
+
+interface LimitByDateQuery {
+  $match: {
+    _id: {
+      [key: string]: Types.ObjectId;
+    };
+  };
+}
+
+const buildLimitByDateQuery = (
```

Comment:
I think the greater than/less than query should be built in its own function, since that's going to be reused. I think of this query as just a special case of a greater than/less than query, the only additional logic we have to do is turning the dates into ObjectIds with datetime. That means we should factor out all the logic that is not specific to turning the dates into an objectId into functions. 

```
buildGtQuery(obj, val) {
   // Allows us to handle the case where val is falsey directly in the query builder, which is what builder functions should do imo
   return !!val ? {...obj, $gt: val} : obj;
}

buildLtQuery(obj, val) {
   return !!val ? {...obj, $gt: val} : obj;
}
```

That means we can simplify this function into:
```
buildLimitByDataQuery(startDate, endDate) {
   const matchQuery = {$match: {_id: {} } };
   builtGtQuery(matchQuery, Types.ObjectId(getDateTime(startDate)));
   builtLtQuery(matchQuery, Types.ObjectId(getDateTime(endDate)));
   return matchQuery;
}
```

Might've missed something here, and the code isn't really proper, but what I'm getting at is that building queries naturally invokes the builder pattern and we should use the builder pattern to its maximum, and imo this is what that looks like.